File: tool/map-editor-cs/MapEditor/IO/CsvMapWriter.cs
using System.Collections.Generic;
using System.IO;
using System.Text;
using tool.mapeditor.model;

namespace tool.mapeditor.io;

public static class CsvMapWriter
{
    public static void Save(EditableL1Map map, string path)
    {
        var directory = Path.GetDirectoryName(path);
        if (!string.IsNullOrEmpty(directory))
        {
            Directory.CreateDirectory(directory);
        }

        var bytes = Encoding.UTF8.GetBytes(map.ToCsv());
        File.WriteAllBytes(path, bytes);
    }

    public static void ExportAll(IEnumerable<EditableL1Map> maps, string directory)
    {
        Directory.CreateDirectory(directory);
        foreach (var map in maps)
        {
            var path = Path.Combine(directory, $"{map.MapId}.txt");
            Save(map, path);
        }
    }
}

File: tool/map-editor/src/main/java/tool/mapeditor/io/CsvMapWriter.java
package tool.mapeditor.io;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

import tool.mapeditor.model.EditableL1Map;

public class CsvMapWriter {
    private final Path mapsDirectory;

    public CsvMapWriter(Path mapsDirectory) {
        this.mapsDirectory = mapsDirectory;
    }

    public void write(EditableL1Map map) throws IOException {
        Path file = mapsDirectory.resolve(map.getId() + ".txt");
        Files.createDirectories(file.getParent());
        byte[] data = map.toCsv().getBytes(StandardCharsets.UTF_8);
        Files.write(file, data);
    }

    public static CsvMapWriter forProjectRoot() {
        return new CsvMapWriter(Paths.get("maps"));
    }
}
